tests/src/Unit/SectionsCollectorTest.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace {
+
+  // Mock a global function.
+  function file_scan_directory($dir) {
+    $path = getcwd() . '/web/modules/contrib/ckeditor5_sections/tests/src/Unit/assets/data/';
+    return [
+      $path . 'test1.yml' =>
+        (object) [
+          'uri' => $path . 'test1.yml',
+          'filename' => 'test1.yml',
+          'name' => 'test1',
+        ],
+      $path . 'test2.yml' =>
+        (object) [
+          'uri' => $path . 'test2.yml',
+          'filename' => 'test2.yml',
+          'name' => 'test2',
+        ],
+    ];
+  }
+
+}
+
+namespace Drupal\Tests\ckeditor5_sections\Unit {
+
+  use Drupal\ckeditor5_sections\SectionsCollector;
+  use Drupal\Tests\UnitTestCase;
+
+  /**
+   * Class SectionsCollectorTest
+   *
+   * @package Drupal\Tests\ckeditor5_sections\Unit
+   */
+  class SectionsCollectorTest extends UnitTestCase {
+
+    public $sectionsCollector;
+
+    protected $entityTypeManager;
+
+    protected $twigProcessor;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setUp() {
+      parent::setUp();
+      $this->entityTypeManager = $this->createMock('Drupal\Core\Entity\EntityTypeManagerInterface');
+      $this->twigProcessor = $this->createMock('Drupal\ckeditor5_sections\TwigProcessor');
```

Comment:
The unit test is fine, but it think we should add a KernelTest that actually uses the processor too. As a layer of securtity. And we will need it anyway as soon as we have to enable twig includes.